### PR TITLE
Update to Electron 7

### PR DESCRIPTION
**Pull request by stackptr**
I was complaining about a system vibrancy problem in #1091 and it was indicated that to fix the issue properly on Catalina it was necessary to wait for the next version of Electron to be released. It was indeed released a week ago.

This PR updates the version and then fixes the errors that I saw running `npm install && npm start`.

(It does not appear to fix my original complaint, any additional work will go in another branch)

### CODE REVIEW

**Comment by stackptr** [comment]
There are some non-blocking warnings that I will also try to address:

```
> tsc && electron .

(electron) 'getName function' is deprecated and will be removed. Please use 'name property' instead.
(electron) 'systemPreferences.isDarkMode()' is deprecated and will be removed. Please use 'nativeTheme.shouldUseDarkColors' instead.
(electron) 'setAppLevelAppearance function' is deprecated and will be removed. Please use 'appLevelAppearance property' instead.
(electron) 'setBadgeCount function' is deprecated and will be removed. Please use 'badgeCount property' instead.
```

**Comment by AidanGG** [comment]
This fixes my issue that I opened up yesterday too https://github.com/sindresorhus/caprine/issues/1153 :smile:.

**Comment by stackptr** [comment]
It's unclear to me how to resolve the lint error on `@typescript-eslint/no-unused-vars` for the `IpcMain` class, but since it's a hack that seems to be resolved in electron/electron#20712 maybe the thing to do is to wait for the minor version that fixes it? I'm not quite sure in what version that would be expected to land.

**Comment by sindresorhus** [comment]
Just use a `// @ts-ignore` comment to work around it.

**Comment by stackptr** [comment]
Dark mode following system appearance doesn't have to go in this PR, but pretty much all the existing logic for it seems to have been deprecated. I've been trying to get this to use the new `nativeTheme` api and I've finally determined that it is not importable in `source/browser.ts` where `setDarkMode()` lives. 

The import statement `import {nativeTheme} from 'electron'` does not yield type errors it just turns out undefined when trying to use it. After some digging, I see that the different "processes" have a module listing in the main electron repo, `lib/${PROCESS}/api/module_list.ts`. So, for `browser.ts` the process is `renderer`, which lacks any mention of the relevant `nativeTheme` api. It is however accessible in `source/index.ts`, which could be the `browser` process. (Which seems to be referred to as Main in the electron docs).

Anyways, it appears some code needs to be shuffled around here. I can send `nativeTheme` in the `set-dark-mode` event in the `webContents` `EventEmitter` but then there is the `setDarkMode` call in the `DOMContentLoaded` event that would also need it. I'll poke around further but any ideas regarding this architecture would be helpful!

**Comment by sindresorhus** [comment]
You could use https://github.com/sindresorhus/electron-util#api-1

File: source/index.ts [hunk 1]
```diff
@@ -156,7 +154,18 @@ ipcMain.on('update-tray-icon', updateTrayIcon);
...
+interface OnSendHeadersDetails {
+	id: number;
+	url: string;
+	method: string;
+	webContentsId?: number;
+	resourceType: string;
+	referrer: string;
+	timestamp: number;
+	requestHeaders: Record<string, string>;
 }
```
Comment (sindresorhus) [inline source/index.ts]: Why do we have to manually type this now?

File: source/emoji.ts [hunk 2]
```diff
@@ -12,6 +12,16 @@ import {memoize} from 'lodash';
...
 
+// TODO: removeListener is typed incorrectly in v7.0.0. Remove on later version
+// Issue: https://github.com/electron/electron/issues/20705
+// PR: https://github.com/electron/electron/pull/20712
+declare module 'electron' {
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	class IpcMain {
+		removeListener(event: string, listener: (...args: any[]) => void): this;
+	}
+}
```
Comment (sindresorhus) [inline source/emoji.ts]: For reference, this is always why I wait a while after a major Electron release before upgrading. There are always silly bugs like this.

File: source/index.ts [hunk 3]
```diff
@@ -444,28 +453,31 @@ function createMainWindow(): BrowserWindow {
...
 		event.preventDefault();
 
 		if (url === 'about:blank') {
 			if (frameName !== 'about:blank') {
 				// Voice/video call popup
 				options.show = true;
 				options.titleBarStyle = 'default';
-				options.webPreferences.nodeIntegration = false;
-				options.webPreferences.preload = path.join(__dirname, 'browser-call.js');
+				if (options.webPreferences !== undefined) {
```
Comment (sindresorhus) [inline source/index.ts]: Why would it be `undefined`?

And even if it is, the correct way would be to do `options.webPreferences = options.webPreferences || {}`.

Comment (stackptr) [reply source/index.ts]: It's marked as optional in the docs: https://electronjs.org/docs/api/browser-window#new-browserwindowoptions

and was throwing a type error without this change. But I think I misunderstood what this code was doing, it appears as though an `options` object is being built so I'll revert and see a better way of doing this

File: source/index.ts (hunk 1 above)
Comment (stackptr) [reply source/index.ts]: It doesn't seem to be present in the `electron` library. I'm not sure why

File: package.json [hunk 4]
```diff
@@ -48,7 +48,7 @@
 		"@typescript-eslint/eslint-plugin": "^2.3.3",
 		"@typescript-eslint/parser": "^2.3.3",
 		"del-cli": "^3.0.0",
-		"electron": "^6.0.11",
+		"electron": "^7.0.0",
```
Comment (sindresorhus) [inline package.json]:
```suggestion
		"electron": "^7.1.1",
```

Is the latest now.

File: source/emoji.ts [hunk 5]
```diff
@@ -12,6 +12,16 @@ import {memoize} from 'lodash';
 import config from './config';
 import {showRestartDialog, getWindow, sendBackgroundAction} from './util';
 
+// TODO: removeListener is typed incorrectly in v7.0.0. Remove on later version
```
Comment (sindresorhus) [inline source/emoji.ts]: I think this is fixed in the latest Electron version.

**Comment by stackptr** [comment]
Addressed merge conflicts, resolved some questions and using later Electron version to avoid that type hack.

**Comment by sindresorhus** [comment]
Thanks 🙌

**Pull request merged**